### PR TITLE
use GH reusable workflows to reduce .yml duplication

### PR DESCRIPTION
**Pull request by adamsitnik**
I am about to introduce more new packages and I wonder whether it would be possible to reduce the .yml duplication by using workflows.

I wrote it based on https://dev.to/n3wt0n/avoid-duplication-github-actions-reusable-workflows-3ae8

### CODE REVIEW

**Comment by adamsitnik** [comment]
@lfraile it's ready for review, PTAL

File: .github/workflows/healthchecks_azurekeyvault_cd_preview.yml [hunk 1]
```diff
@@ -8,23 +8,9 @@ on:
 
 jobs:
   build:
-    env:
+    uses: ./.github/workflows/reusable_cd_preview_workflow.yml
```
Comment (lfraile) [inline .github/workflows/healthchecks_azurekeyvault_cd_preview.yml:11]:
```suggestion
    uses: ./.github/workflows/reusable_cd_preview_workflow.yml
    secrets: inherit
```
We will need this so we can use ${{secrets.NUGET_API_KEY}}

File: .github/workflows/healthchecks_azurekeyvault_cd.yml [hunk 2]
```diff
@@ -8,22 +8,8 @@ on:
 
 jobs:
   build:
-    env:
+    uses: ./.github/workflows/reusable_cd_workflow.yml
```
Comment (lfraile) [inline .github/workflows/healthchecks_azurekeyvault_cd.yml:11]:
```suggestion
    uses: ./.github/workflows/reusable_cd_workflow.yml
    secrets: inherit
```
We will need this so we can use ${{secrets.NUGET_API_KEY}}

**Review by lfraile — CHANGES_REQUESTED** [review]

**Comment by adamsitnik** [comment]
@lfraile great catch! I've applied your suggestion to all files, PTAL

**Review by lfraile — APPROVED** [review]

**Comment by sungam3r** [comment]
@adamsitnik Thank you very much for that!

**Pull request merged**